### PR TITLE
win: Allow UNC path with forward slash

### PR DESCRIPTION
**Pull request by earnal**
PR to fix the issue #3159.
The origin [issue](https://gitlab.kitware.com/cmake/cmake/-/issues/22141) is on CMake side: **CTest cannot run executable from network path**.

### CODE REVIEW

**Comment by mathstuf** [comment]
Bump. How to get maintainer attention on this?

File: src/win/process.c
Comment (vtjnash) [inline src/win/process.c:174]:
```suggestion
	(dir[0] == L'\\' && dir[0] == L'/') ||
	(dir[1] == L'\\' && dir[1] == L'/') 
```

**Review by vtjnash — COMMENTED** [review]
Sorry, I only comment on the issue, and perhaps should have asked for the same change here. Otherwise LGTM

File: src/win/process.c
Comment (earnal) [reply src/win/process.c:174]: @vtjnash I cannot approve your suggestion.
`(dir[0] == L'\\' && dir[0] == L'/')` will never be true... idem for `dir[1]`.

Comment (vtjnash) [reply src/win/process.c:174]:
```suggestion
	(dir[0] == L'\\' || dir[0] == L'/') &&
	(dir[1] == L'\\' || dir[1] == L'/') 
```

Comment (vtjnash) [reply src/win/process.c:174]: bump

**Comment by earnal** [comment]
There is not much responsiveness in this repository... 
I will try to push this fix directly on CMake where things move faster.

File: src/win/process.c
Comment (earnal) [reply src/win/process.c:174]: @vtjnash your second suggestion will allow paths starting with "\/" and "/\".
I think that windows only support "//" and "\\".

Comment (vtjnash) [reply src/win/process.c:174]: Those paths worked when I tested windows

Comment (vtjnash) [reply src/win/process.c:174]: If you are able to test and confirm which is correct, I can probably get this merged before the current release

File: src/win/process.c
Comment (vtjnash) [inline src/win/process.c:175]:
```suggestion
  if (dir_len > 2 &&
      ((dir[0] == L'\\' || dir[0] == L'/') &&
       (dir[1] == L'\\' || dir[1] == L'/')) {
```

**Review by vtjnash — APPROVED** [review]
LGTM, with style nit fixed when landing

CI: https://ci.nodejs.org/job/libuv-test-commit/2187/

File: src/win/process.c
Comment (vtjnash) [inline src/win/process.c:174]:
```suggestion
       (dir[1] == L'\\' || dir[1] == L'/'))) {
```

**Comment by vtjnash** [comment]
CI: https://ci.nodejs.org/job/libuv-test-commit/2191/

**Comment by vtjnash** [comment]
Half the build bots were/are broken, so I reran, and am hoping to get functional ones this time: https://ci.nodejs.org/job/libuv-test-commit-windows-cmake/386/

**Comment by vtjnash** [comment]
Let's try again now: https://ci.nodejs.org/job/libuv-test-commit-windows-cmake/389/

**Comment by earnal** [comment]
Bump. How to get maintainer attention on this?

**Pull request merged**
